elastica/rod/knot_theory.py

```diff
@@ -38,15 +39,15 @@ def radius(self) -> np.ndarray: ...
     def base_length(self) -> np.ndarray: ...
 
 
-class KnotTheory:
+class KnotTheory(KnotTheoryCompatibleProtocol):
```

Comment:
```suggestion
class KnotTheory:
```

Reply:
No need to inherit protocol here. Also because of this, unit-tests are not passing.